test/src/materials/InterfaceValueMaterial.C

```diff
@@ -0,0 +1,83 @@
+InputParameters
+validParams<InterfaceValueMaterial>()
+{
+  InputParameters params = validParams<InterfaceMaterial>();
+  params.addClassDescription("Calculates a variable's jump value across an interface.");
+  params.addRequiredParam<std::string>("mat_prop_master",
+                                       "The material property on the master side of the interface");
+  params.addRequiredParam<std::string>("mat_prop_slave",
+                                       "The material property on the slave side of the interface");
+  params.addRequiredParam<std::string>("mat_prop_out_basename",
+                                       "The base name for the output material property");
+  params.addRequiredCoupledVar("var_master",
+                               "The material property on the master side of the interface");
+  params.addRequiredCoupledVar("var_slave",
+                               "The material property on the slave side of the interface");
+  params.addRequiredParam<std::string>("mat_prop_var_out_basename",
+                                       "The base name for the output material property");
+  params.addParam<MooseEnum>("interface_value_type",
+                             InterfaceValueTools::InterfaceAverageOptions(),
+                             "Type of scalar output");
+  return params;
+}
+
+InterfaceValueMaterial::InterfaceValueMaterial(const InputParameters & parameters)
+  : InterfaceMaterial(parameters),
+    _mp_master_name(getParam<std::string>("mat_prop_master")),
+    _mp_slave_name(getParam<std::string>("mat_prop_slave")),
+    _mp_master(getMaterialPropertyByName<Real>(_mp_master_name)),
+    _mp_slave(getNeighborMaterialProperty<Real>(_mp_slave_name)),
+    _var_master(coupledValue("var_master")),
+    _var_slave(coupledNeighborValue("var_slave")),
+
+    _interface_value_type(parameters.get<MooseEnum>("interface_value_type")),
+    _mp_out_base_name(getParam<std::string>("mat_prop_out_basename")),
+    _mp_var_out_base_name(getParam<std::string>("mat_prop_var_out_basename")),
+    _interface_value(
+        declareProperty<Real>(_mp_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_2(
+        declareProperty<Real>(_mp_var_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_old(
+        getMaterialPropertyOld<Real>(_mp_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_2_old(getMaterialPropertyOld<Real>(_mp_var_out_base_name + "_" +
+                                                        std::string(_interface_value_type)))
+{
+}
+
+void
+InterfaceValueMaterial::computeQpProperties()
+{
+  // TODO remove when everything works correctly
+  if (_mp_master[_qp] != _var_master[_qp])
+    mooseWarning("the material property and variable values on the master side do not coincide.");
+  if (_mp_slave[_qp] != _var_slave[_qp])
+    mooseWarning("the material property and variable values on the slave side do not coincide.");
```

Comment:
@lindsayad and @andrsd I need some help here.
I can't figure out why I can nicely couple neighbour variable but i can't do the same whit material properties. The coupled neighbor material property always return zero value `getNeighborMaterialProperty` always has a zero value.
The call starts from `ComputeElemAuxBcsThread.C`